R/install.R

```diff
@@ -128,3 +152,82 @@ cmdstan_git_checkout_branch <- function(repo_branch,
   )
   invisible(install_log)
 }
+
+
+
+# internal ----------------------------------------------------------------
+
+# construct url for download from cmdstan version number
+github_download_url <- function(version_number) {
+  base_url <- "https://github.com/stan-dev/cmdstan/releases/download/"
+  paste0(base_url, "v", version_number,
+         "/cmdstan-", version_number, ".tar.gz")
+}
+
+# get version number of latest release
+latest_released_version <- function() {
+  if (!requireNamespace("jsonlite", quietly = TRUE)) {
+    stop("Please install the jsonlite package.", call. = FALSE)
+  }
+  contents <- url("https://api.github.com/repos/stan-dev/cmdstan/releases/latest")
+  tag_name <- jsonlite::parse_json(contents)[["tag_name"]]
+  version_number <- sub("v", "", tag_name)
+  version_number
+}
+
+# internal functions to run system commands -------------------------------
+clone_repo <- function(dir, repo_url, repo_branch, quiet) {
+  git_args <- c("clone", "--recursive",
+                repo_url, paste0("-b", repo_branch),
+                dir)
+  processx::run(
+    command = "git",
+    args = git_args,
+    echo_cmd = FALSE,
+    error_on_status = TRUE,
+    echo = !quiet,
+    spinner = quiet
+  )
+}
+
+build_cmdstan <- function(dir, cores, quiet) {
+  processx::run(
+    make_cmd(),
+    args = c(paste0("-j", cores), "build"),
+    wd = dir,
+    echo_cmd = FALSE,
+    echo = !quiet,
+    spinner = quiet,
+    error_on_status = FALSE,
+    stderr_line_callback = function(x,p) { if(quiet) message(x) },
+    timeout = 600 # timeout after 10 minutes to avoid infinite loop problem?
+  )
+}
+
+build_example <- function(dir, cores, quiet) {
+  processx::run(
+    make_cmd(),
+    args = c(paste0("-j", cores), cmdstan_ext("examples/bernoulli/bernoulli")),
+    wd = dir,
+    echo_cmd = FALSE,
+    echo = !quiet,
+    spinner = quiet,
+    error_on_status = FALSE,
+    stderr_line_callback = function(x,p) { if(quiet) message(x) }
+  )
+}
+
+build_status_ok <- function(process_log, quiet = FALSE) {
+  if (is.na(process_log$status) || process_log$status != 0) {
+    if (!quiet) {
+      suggestion <- "See the error message(s) above."
+    } else {
+      suggestion <- "Please try again with 'quiet=FALSE' to get the full error output."
```

Comment:
The error messages will be output here in any case because of the 

    stderr_line_callback = function(x,p) { if(quiet) message(x) }

So maybe just write out "suggestion <- "See the error message(s) above or try again with 'quiet=FALSE' to get the full installation output."